### PR TITLE
Added a tip about the TMPDIR env variable

### PR DESCRIPTION
**Pull request by javiereguiluz**
This will help solve issues like https://github.com/symfony/symfony/issues/20423

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
We should mention that this doesn't apply to Windows.

**Comment by nicolas-grekas** [comment]
From [msdn](https://msdn.microsoft.com/fr-fr/library/windows/desktop/aa364992(v=vs.85).aspx) for Windows, this var is called `TMP`, `TEMP` or `USERPROFILE` (in this order).

**Comment by javiereguiluz** [comment]
According to the PHP 7 source code, the variables checked on Windows are ` TMP` and `TEMP` (in this order). I can't see `USERPROFILE`  See http://lxr.php.net/xref/PHP-7.0/main/php_open_temporary_file.c#199

**Comment by nicolas-grekas** [comment]
I'd better trust MSDN than a comment... especially since the code is calling `GetTempPath`.

File: components/filesystem/lock_handler.rst [hunk 1]
```diff
@@ -48,9 +48,19 @@ the file, so you can pass any value for this argument.
...
-second argument of the constructor.
+By default, the lock will be created in the system's temporary directory, but
+you can optionally select the directory where locks are created by passing it as
+the second argument of the constructor.
+
+.. tip::
+
+    Another way to configure the directory where the locks are created is to
+    define a special environment variable, because PHP will use that value to
+    override the default temporary directory. In Unix-based systems, define the
```
Comment (xabbuh) [inline components/filesystem/lock_handler.rst]: On Unix-based systems [...]?

File: components/filesystem/lock_handler.rst [hunk 2]
```diff
@@ -48,9 +48,19 @@ the file, so you can pass any value for this argument.
...
+    ``TMPDIR`` variable. In Windows systems, define any of these variables:
```
Comment (xabbuh) [inline components/filesystem/lock_handler.rst]: Same here (On Windows Systems [...])?

**Comment by xabbuh** [comment]
:+1:

Status: Reviewed

**Comment by xabbuh** [comment]
Thank you Javier.

**Pull request closed without merging**